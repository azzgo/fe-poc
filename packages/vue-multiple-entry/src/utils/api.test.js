import apiClient from './api'

beforeEach(() => {
  global.mockApi.reset()
})

test('api work as expected', async () => {
  global.mockApi.onAny().reply(500)

  const request = () => apiClient.get('/some-things')
  expect(Promise.all([request(), request()])).rejects.toThrowErrorMatchingSnapshot()
})

test('api work as expected 2', async () => {
  global.mockApi
    .onAny()
    .replyOnce(200, { data: 'data1' })
    .onAny()
    .replyOnce(200, { data: 'data2' })

  const request = () => apiClient.get('/some-things')

  expect(Promise.all([request(), request()])).resolves.toMatchSnapshot()
})
