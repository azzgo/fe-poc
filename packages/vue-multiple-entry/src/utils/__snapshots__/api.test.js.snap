// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api work as expected 2 1`] = `"Request failed with status code 500"`;

exports[`api work as expected 2 2`] = `
Array [
  Object {
    "config": Object {
      "baseURL": "http://localhost:3000",
      "data": undefined,
      "headers": Object {
        "Accept": "application/json, text/plain, */*",
      },
      "maxContentLength": -1,
      "method": "get",
      "timeout": 0,
      "transformRequest": Object {
        "0": [Function],
      },
      "transformResponse": Object {
        "0": [Function],
      },
      "url": "/some-things",
      "validateStatus": [Function],
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
    },
    "data": Object {
      "data": "data1",
    },
    "headers": undefined,
    "status": 200,
  },
  Object {
    "config": Object {
      "baseURL": "http://localhost:3000",
      "data": undefined,
      "headers": Object {
        "Accept": "application/json, text/plain, */*",
      },
      "maxContentLength": -1,
      "method": "get",
      "timeout": 0,
      "transformRequest": Object {
        "0": [Function],
      },
      "transformResponse": Object {
        "0": [Function],
      },
      "url": "/some-things",
      "validateStatus": [Function],
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
    },
    "data": Object {
      "data": "data2",
    },
    "headers": undefined,
    "status": 200,
  },
]
`;
